src/main/java/org/opentripplanner/routing/edgetype/StreetEdge.java

```diff
@@ -1354,14 +1355,39 @@ protected void calculateLengthFromGeometry () {
                 e.setFloatingVehicleDropoffSuitability(getFloatingVehicleDropoffSuitability());
             }
         } else {
-            if (((TemporarySplitterVertex) v).isEndVertex()) {
-                e1 = new TemporaryPartialStreetEdge(this, (StreetVertex) fromv, v, geoms.first, name);
-                e1.setNoThruTraffic(this.isNoThruTraffic());
-                e1.setStreetClass(this.getStreetClass());
+            if (createSemiPermanentEdges) {
```

Comment:
Should this also check if `this` is a `SemiPermanentPartialStreetEdge`? There's not a reason to split a `SemiPermanentPartialStreetEdge` is there?

Reply:
That check already occurs here: https://github.com/ibi-group/OpenTripPlanner/blob/a29d663dabef6e91d1bc2c079af9c02443dcc50b/src/main/java/org/opentripplanner/routing/edgetype/SemiPermanentPartialStreetEdge.java#L79-L87